pydantic/_internal/_fields.py

```diff
@@ -102,6 +102,15 @@ def collect_model_fields(  # noqa: C901
     BaseModel = import_cached_base_model()
     FieldInfo_ = import_cached_field_info()
 
+    dataclass_fields = {
+        field.name for base in bases for field in (dataclasses.fields(base) if dataclasses.is_dataclass(base) else ())
+    }
```

Comment:
I believe this represents a bit of a logical change.

Previously, we only collected the dc fields for each and every base individually, now we aggregate them all into one dict. Might there be unintended consequences here?

Reply:
Hum maybe we could be even smarter:

```python
for base in bases:
    if hasattr(base, ann_name):
        ...
        if dataclasses.is_dataclass(base):
            continue

Reply:
It might be that it's functionally different but I doubt it. Reverted because this was just logic cleaning (performance is marginal) and let's make sure it doesn't break anything